sw/device/lib/base/mmio.h

```diff
@@ -288,6 +288,30 @@ inline void mmio_region_nonatomic_set_bit32(mmio_region_t base,
   mmio_region_nonatomic_set_mask32(base, offset, 0x1, bit_index);
 }
 
+/**
+ * Copies a block of memory from MMIO to main memory while ensuring that MMIO
+ * accesses are word-aligned.
+ *
+ * @param base the MMIO region to read from.
+ * @param offset the offset to start reading from, in bytes.
+ * @param buf the main memory location to start writing to.
+ * @param count number of bytes to copy.
+ */
+void mmio_region_memcpy_from_mmio32(mmio_region_t base, uint32_t offset,
+                                    void *dst, size_t count);
```

Comment:
Nit: Might as well be consistent with our memcpy, which names these parameters `dest` and `len`.

Reply:
Done.